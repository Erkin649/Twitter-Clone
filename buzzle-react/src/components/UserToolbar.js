import React from 'react';
import { Link } from "react-router-dom";

const UserToolbar = ({ user, logoutUser }) =>
    user ?
    <form action="/logout" onSubmit={event => { event.preventDefault(); logoutUser() }} method="POST">
        <input
            id="logout"
            className="btn btn-outline-light"
            type="submit"
            value={`Logout (@${user.login})`} />
    </form>
    :
    <div className="btn-group">
        <Link className="btn btn-outline-light" to="/login">Login</Link>
        <Link className="btn btn-outline-light" to="/register">Register</Link>
    </div>

export default UserToolbar;
