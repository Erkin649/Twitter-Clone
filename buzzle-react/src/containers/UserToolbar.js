import { connect } from 'react-redux';

import { logoutUser } from '../actions/user.js';
import UserToolbar from '../components/UserToolbar.js';

const mapStateToProps = state => ({
    'user': state.user
});

const mapDispatchToProps = dispatch => ({
    'logoutUser': () => dispatch(logoutUser())
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(UserToolbar);
