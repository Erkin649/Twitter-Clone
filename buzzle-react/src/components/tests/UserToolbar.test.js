import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import UserToolbar from '../UserToolbar.js';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Router><UserToolbar /></Router>, div);
});

it('renders correctly and equal to snapshot', () => {
    const tree = renderer.create(<Router><UserToolbar /></Router>)
                         .toJSON();

    expect(tree).toMatchSnapshot();
});