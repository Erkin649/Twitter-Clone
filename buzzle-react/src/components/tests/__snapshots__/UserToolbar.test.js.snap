// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and equal to snapshot 1`] = `
<div
  className="btn-group"
>
  <a
    className="btn btn-outline-light"
    href="/login"
    onClick={[Function]}
  >
    Login
  </a>
  <a
    className="btn btn-outline-light"
    href="/register"
    onClick={[Function]}
  >
    Register
  </a>
</div>
`;
